homeassistant/components/bluesound/coordinator.py

```diff
@@ -0,0 +1,178 @@
+        return self.data
+
+    def _set_offline(self) -> None:
+        _LOGGER.error(
+            "Node %s is offline or threw an unexpected error, retrying later",
+            self.data.sync_status.id,
+        )
+        self.async_set_updated_data(
+            replace(
+                self.data,
+                is_online=False,
+            )
+        )
+
+    async def _poll_presets_and_inputs_loop(self) -> None:
+        while True:
+            await asyncio.sleep(PRESET_AND_INPUTS_INTERVAL.total_seconds())
+            try:
+                preset = await self.player.presets()
+                inputs = await self.player.inputs()
+                self.async_set_updated_data(
+                    replace(
+                        self.data,
+                        presets=preset,
+                        inputs=inputs,
+                        is_online=True,
+                    )
+                )
+            except PlayerUnreachableError:
+                self._set_offline()
+            except asyncio.CancelledError:
+                return
+            except:  # noqa: E722 - this loop should never stop
+                self._set_offline()
+
+    async def _poll_status_loop(self) -> None:
+        """Loop which polls the status of the player."""
+        while True:
+            try:
+                status = await self.player.status(
+                    etag=self.data.status.etag, poll_timeout=120, timeout=125
+                )
+                self.async_set_updated_data(
+                    replace(
+                        self.data,
+                        status=status,
+                        is_online=True,
+                    )
+                )
+            except PlayerUnreachableError:
+                self._set_offline()
+                await asyncio.sleep(NODE_OFFLINE_CHECK_TIMEOUT.total_seconds())
+            except asyncio.CancelledError:
+                return
+            except:  # noqa: E722 - this loop should never stop
+                self._set_offline()
+                await asyncio.sleep(NODE_OFFLINE_CHECK_TIMEOUT.total_seconds())
+
+    async def _poll_sync_status_loop(self) -> None:
+        """Loop which polls the sync status of the player."""
+        while True:
+            try:
+                sync_status = await self.player.sync_status(
+                    etag=self.data.sync_status.etag, poll_timeout=120, timeout=125
+                )
+                self.async_set_updated_data(
+                    replace(
+                        self.data,
+                        sync_status=sync_status,
+                        is_online=True,
+                    )
+                )
+            except PlayerUnreachableError:
+                self._set_offline()
+                await asyncio.sleep(NODE_OFFLINE_CHECK_TIMEOUT.total_seconds())
+            except asyncio.CancelledError:
+                raise
+            except:  # noqa: E722 - all errors must be caught for this loop
```

Comment:
Why not `except Exception`

Reply:
I can do that, but it leads to another linting error. Is there any difference between the two?

Reply:
Nevermind. That one solves itself when using `async_set_update_error`.